segfast/segyio_loader.py

```diff
@@ -125,32 +123,60 @@ def load_headers(self, headers, reconstruct_tsf=True, sort_columns=True, tracewi
+        headers = self._make_headers_specs(headers)
+        if not all([header.is_standard_except_name for header in headers]):
+            warnings.warn("All nonstandard trace headers byte positions and dtypes will be ignored.")
 
-        headers_bytes = self.headers_to_bytes(headers)
+        if reconstruct_tsf:
+            headers = [header for header in headers if header.name != 'TRACE_SEQUENCE_FILE']
 
         # Load data to buffer
         buffer = np.empty((self.n_traces, len(headers)), dtype=np.int32)
         if tracewise:
             for i, header_ in Notifier(pbar, total=self.n_traces, frequency=1000)(enumerate(self.file_handler.header)):
-                for j, byte_ in enumerate(headers_bytes):
-                    buffer[i, j] = header_.getfield(header_.buf, byte_)
+                for j, header in enumerate(headers):
+                    buffer[i, j] = header_.getfield(header_.buf, header.start_byte)
         else:
             for i, header in enumerate(headers):
                 buffer[:, i] = self.load_header(header)
 
         # Convert to pd.DataFrame, optionally add TSF and sort
-        dataframe = pd.DataFrame(buffer, columns=headers, copy=False)
+        dataframe = pd.DataFrame(buffer, columns=[item.name for item in headers], copy=False)
         dataframe = self.postprocess_headers_dataframe(dataframe, headers=headers,
                                                        reconstruct_tsf=reconstruct_tsf, sort_columns=sort_columns)
         return dataframe
 
+    def _make_headers_specs(self, headers):
+        """ Make instances of TraceHeader. """
+        byteorder = self.ENDIANNESS_TO_SYMBOL[self.endian]
+
+        if headers == 'all':
+            return [TraceHeaderSpec(start_byte, byteorder=byteorder)
+                    for start_byte in TraceHeaderSpec.STANDARD_BYTE_TO_HEADER]
+
+        headers_ = []
+        for header in headers:
+            if isinstance(header, TraceHeaderSpec):
+                headers_.append(header.set_default_byteorder(byteorder))
+            else:
+                if isinstance(header, str):
+                    init_kwargs = {'name': header}
+                elif isinstance(header, int):
+                    init_kwargs = {'start_byte': header}
+                elif isinstance(header, tuple):
+                    arg_name = 'name' if isinstance(header[0], str) else 'start_byte'
+                    init_kwargs = {arg_name: header[0], 'dtype': header[1]}
+                elif isinstance(header, dict):
+                    init_kwargs = header
+                init_kwargs = {'byteorder': byteorder, **init_kwargs}
+                headers_.append(TraceHeaderSpec(**init_kwargs))
+        return headers_
+
     def load_header(self, header):
         """ Read one header from the file. """
-        return self.file_handler.attributes(getattr(segyio.TraceField, header))[:]
+        if isinstance(header, (int, str)):
+            header = TraceHeaderSpec(header)
```

Comment:
Just to unify parsing logic with `load_headers`: now they work differently.
```suggestion
        header = self._make_headers_specs([header])[0]
```